### PR TITLE
Feat / Vitest axe WCAG tests for components

### PR DESCRIPTION
**Pull request by langemike**
This change will automatically asses our components to the WCAG 2.1 and 2.2 guidelines. Having this in place makes sure our current state is according to the specs and it will prevent us from accidentally breaking accessibility. 

We've used Vitest Axe plugin which internally uses a plugin from Axe [axe-core](https://www.npmjs.com/package/axe-core). Axe is an acclaimed party when it comes to development tools related to web accessibility. Potentially more integrations from Axe will follow, with this being the first.

It also contains a minor change for the `Button` component to make it succeed with the WCAG guidelines.

Axe docs for reference: https://www.deque.com/axe/core-documentation/api-documentation/
Based on the work of: https://github.com/Videodock/ott-web-app/pull/101
Ticket: https://videodock.atlassian.net/browse/OTT-1133

### CODE REVIEW

File: packages/ui-react/package.json [hunk 1]
```diff
@@ -45,7 +45,8 @@
     "typescript-plugin-css-modules": "^5.0.2",
     "vi-fetch": "^0.8.0",
     "vite-plugin-svgr": "^4.2.0",
-    "vitest": "^1.3.1"
+    "vitest": "^1.3.1",
+    "vitest-axe": "^1.0.0-pre.3"
```
Comment (AntonLantukh) [inline packages/ui-react/package.json:50]: @langemike do you feel confident about this one?

File: packages/ui-react/src/components/Shelf/Shelf.test.tsx [hunk 2]
```diff
@@ -93,4 +94,36 @@ describe('Shelf Component tests', () => {
...
+          title="Test Shelf"
+          type={'playlist'}
+          accessModel={'AVOD'}
+          hasSubscription={true}
+          isLoggedIn={true}
+          playlist={playlist}
+          enableCardTitles
+          enableTitle
+        />
+        ,<h2>Featured shelf</h2>
```
Comment (ChristiaanScheermeijer) [inline packages/ui-react/src/components/Shelf/Shelf.test.tsx]:
```suggestion
        <h2>Featured shelf</h2>
```

**Review by ChristiaanScheermeijer — APPROVED** [review]
Nice changes @langemike 🙌 !

File: packages/ui-react/package.json (hunk 1 above)
Comment (langemike) [reply packages/ui-react/package.json:50]: I sure am!

**Review by AntonLantukh — APPROVED** [review]

**Comment by AntonLantukh** [comment]
@langemike could you please get changes from develop in this branch?

File: packages/ui-react/src/components/Shelf/Shelf.test.tsx (hunk 2 above)
Comment (langemike) [reply packages/ui-react/src/components/Shelf/Shelf.test.tsx]: fixed

**Comment by langemike** [comment]
@AntonLantukh I rebased the changes from develop-branch and fixed some failing unit tests. should be all good now

**Pull request merged**